### PR TITLE
Remove FairlearnDashboard and replace it with matplotlib-based plots

### PR DESCRIPTION
**Pull request by romanlutz**
With a slight delay (originally targeted for April) I'm finally removing the `FairlearnDashboard` since a newer version already exists in `raiwidgets`. The documentation is updated to instead use the plots @MiroDudik created with a single line directly from the `MetricFrame`. In the future we want to add more kinds of plots as already mentioned in #758 #666 and #668 . Specifically, the model comparison plots do not yet have a replacement yet.

Note that the "example" added to the `examples` directory is not shown under "Example notebooks" on the webpage, which is intentional since it's technically not a notebook.

This also makes #561 mostly redundant, which I'll close shortly. 
#667 is also directly addressed with this PR as the examples illustrate.

### CODE REVIEW

File: CODEOWNERS.md [hunk 1]
```diff
@@ -5,7 +5,6 @@
 /fairlearn/metrics/ @riedgar-ms @rihorn2
```
Comment (romanlutz) [inline CODEOWNERS.md:5]: I realize this file needs more updates, but that's for a separate PR.

File: examples/plot_grid_search_census.py [hunk 2]
```diff
@@ -107,11 +99,16 @@
...
-                   y_true=Y_test,
-                   y_pred={"unmitigated": unmitigated_predictor.predict(X_test)})
+# We can start to assess the predictor's fairness using the `MetricFrame`:
+metric_frame = MetricFrame(metric={"accuracy": skm.accuracy_score, "selection_rate": selection_rate},
+                           sensitive_features=A_test,
+                           y_true=Y_test,
+                           y_pred=unmitigated_predictor.predict(X_test))
+print(metric_frame.overall)
+print(metric_frame.by_group)
+metric_frame.by_group.plot.bar(
```
Comment (romanlutz) [inline examples/plot_grid_search_census.py:109]: renders as follows: 
image

File: examples/plot_grid_search_census.py [hunk 3]
```diff
@@ -193,19 +190,28 @@ def classifier(X): return m.predict(X)
...
+                                     y_pred=predictions[key])
+
+import matplotlib.pyplot as plt
+x = [metric_frame.overall['accuracy'] for metric_frame in metric_frames.values()]
+y = [metric_frame.difference()['selection_rate'] for metric_frame in metric_frames.values()]
+keys = list(metric_frames.keys())
+plt.scatter(x, y)
+for i in range(len(x)):
+    plt.annotate(keys[i], (x[i] + 0.0003, y[i]))
+plt.xlabel("accuracy")
```
Comment (romanlutz) [inline examples/plot_grid_search_census.py:213]: renders as follows:
image

File: docs/quickstart.rst [hunk 4]
```diff
@@ -134,37 +132,45 @@ their label:
...
+            <matplotlib.axes._subplots.AxesSubplot object at 0x0...>],
+           [<matplotlib.axes._subplots.AxesSubplot object at 0x0...>,
+            <matplotlib.axes._subplots.AxesSubplot object at 0x0...>,
+            <matplotlib.axes._subplots.AxesSubplot object at 0x0...>],
+           [<matplotlib.axes._subplots.AxesSubplot object at 0x0...>,
+            <matplotlib.axes._subplots.AxesSubplot object at 0x0...>,
+            <matplotlib.axes._subplots.AxesSubplot object at 0x0...>]],
+          dtype=object)
+
+.. figure:: auto_examples/images/sphx_glr_plot_quickstart_001.png
```
Comment (romanlutz) [inline docs/quickstart.rst:170]: renders as follows:
image

File: docs/user_guide/assessment.rst [hunk 5]
```diff
@@ -394,15 +394,45 @@ For more examples, please
...
+        'false positive rate': false_positive_rate,
+        'true positive rate': true_positive_rate,
+        'selection rate': selection_rate,
+        'count': lambda y_true, y_pred: y_true.shape[0]}
+    metric_frame = MetricFrame(metrics, y_true, y_pred, sensitive_features=sex)
+    metric_frame.by_group.plot.bar(
+        subplots=True, layout=[3,3], legend=False, figsize=[12,8],
+        title='Show all metrics')
+
+.. figure:: ../auto_examples/images/sphx_glr_plot_quickstart_001.png
```
Comment (romanlutz) [inline docs/user_guide/assessment.rst:425]: renders as follows: 
image

File: scripts/build_documentation.py [hunk 6]
```diff
@@ -33,7 +33,7 @@
 
 
 def _build_argument_parser():
-    desc = "Build widget for fairlearn dashboard"
+    desc = "Build documentation for Fairlearn"
```
Comment (romanlutz) [inline scripts/build_documentation.py:36]: This was previously mislabeled (probably a copy-paste error).

**Comment by adrinjalali** [comment]
Could we please have two PRs, one to replace the dashboard with other plotting functionality, and then remove the dashboard itself? Much easier to review.

**Comment by romanlutz** [comment]
> Could we please have two PRs, one to replace the dashboard with other plotting functionality, and then remove the dashboard itself? Much easier to review.

Fair! I was thinking about it the other way round, but that wouldn't have work since it leaves massive holes all over the documentation. The way you suggest is preferable! I'll leave this as the removal PR.

**Comment by romanlutz** [comment]
> > Could we please have two PRs, one to replace the dashboard with other plotting functionality, and then remove the dashboard itself? Much easier to review.
> 
> Fair! I was thinking about it the other way round, but that wouldn't have work since it leaves massive holes all over the documentation. The way you suggest is preferable! I'll leave this as the removal PR.

I created #770 for this purpose now. Let's leave this one open at least until #770 is complete. Thanks!

**Review by hildeweerts — APPROVED** [review]
LGTM but I would really like @riedgar-ms or @MiroDudik to have look as well, because I'm not that familiar with the widget part of the library :)

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**